File: Digital Business Enablement/bluegather/src/main/java/br/com/fiap/bluegather/service/TipoEventoService.java
package br.com.fiap.bluegather.service;

import br.com.fiap.bluegather.dto.TipoEventoDTO;
import br.com.fiap.bluegather.model.TipoEvento;
import br.com.fiap.bluegather.repository.TipoEventoRepository;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.server.ResponseStatusException;

import java.util.Objects;
import java.util.Set;
import java.util.stream.Collectors;

@Service
public class TipoEventoService {

    @Autowired
    private TipoEventoRepository tipoEventoRepository;

    public Page<TipoEventoDTO> listAll(Pageable pageRequest) {
        Page<TipoEvento> list = tipoEventoRepository.findAll(pageRequest);
        return list.map(this::convertToDto);
    }

    public TipoEventoDTO findById(Long id) {
        TipoEvento entity = findEntityById(id);
        return convertToDto(entity);
    }

    @Transactional
    public TipoEventoDTO create(TipoEventoDTO newData) {
        TipoEvento entity = convertToEntity(newData);
        TipoEvento savedEntity = tipoEventoRepository.save(entity);
        return convertToDto(savedEntity);
    }

    @Transactional
    public TipoEventoDTO update(Long id, TipoEventoDTO updatedData) {
        findEntityById(id);
        updatedData.setId(id);
        TipoEvento updatedEntity = convertToEntity(updatedData);    
        TipoEvento savedEntity = tipoEventoRepository.save(updatedEntity);
        return convertToDto(savedEntity);
    }
    
    @Transactional
    public void delete(Long id) {
        TipoEvento entity = findEntityById(id);
        tipoEventoRepository.delete(entity);
    }

    public TipoEvento findEntityById(Long id) {
        return tipoEventoRepository.findById(id)
                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "(" + getClass().getSimpleName() + ") - TipoEvento não encontrado(a) por ID: " + id));
    }
    
    public Set<TipoEventoDTO> findByNomeContainingIgnoreCase(String nome) {
        Set<TipoEvento> list = tipoEventoRepository.findByNomeContainingIgnoreCase(nome);
        return list.stream().map(this::convertToDto).collect(Collectors.toSet());
    }
    
    public TipoEventoDTO convertToDto(TipoEvento entity) {
        TipoEventoDTO dto = new TipoEventoDTO();
        dto.setId(entity.getId());
        dto.setNome(entity.getNome());
        return dto;
    }

    public TipoEvento convertToEntity(TipoEventoDTO dto) {
        if (Objects.isNull(dto)) {
            throw new IllegalArgumentException("(" + getClass().getSimpleName() + ") - TipoEventoDTO não pode ser nulo.");
        }
        TipoEvento entity;
        if (dto.getId() != null) {
            entity = findEntityById(dto.getId());
        } else {
            entity = new TipoEvento();
        }
        entity.setNome(dto.getNome());
        return entity;
    }
}

File: Enterprise Application Development/bluegather/Services/TipoEventoService.cs
using Microsoft.EntityFrameworkCore;
using BlueGather.Dto;
using BlueGather.Models;
using BlueGather.Context;

namespace BlueGather.Services;
public class TipoEventoService
{
    private readonly BlueGatherContext _context;

    public TipoEventoService(BlueGatherContext context)
    {
        _context = context;
    }

    public async Task<List<TipoEventoDto>> FindAll()
    {
        var list = await _context.TipoEvento
            .ToListAsync();

        return list.Select(entity => ConvertToDto(entity)).ToList();
    }

    public async Task<TipoEventoDto> FindById(long id)
    {
        var entity = await FindEntityById(id);
        return ConvertToDto(entity);
    }

    public async Task<TipoEventoDto> Create(TipoEventoDto newData)
    {
        var entity = await ConvertToEntity(newData);
        _context.TipoEvento.Add(entity);
        await _context.SaveChangesAsync();
        return ConvertToDto(entity);
    }

    public async Task<TipoEventoDto> Update(long id, TipoEventoDto updatedData)
    {
        var entity = await FindEntityById(id);
        updatedData.Id = entity.Id;
        var updatedEntity = await ConvertToEntity(updatedData);
        _context.Entry(entity).CurrentValues.SetValues(updatedEntity);
        await _context.SaveChangesAsync();
        return ConvertToDto(entity);
    }

    public async Task Delete(long id)
    {
        var entity = await FindEntityById(id);
        _context.TipoEvento.Remove(entity);
        await _context.SaveChangesAsync();
    }

    public async Task<TipoEventoModel> FindEntityById(long id)
    {
        var entity = await _context.TipoEvento
            .FirstOrDefaultAsync(x => x.Id == id);
        if (entity == null)
        {
            throw new KeyNotFoundException($"Objeto não encontrado com o ID: {id}");
        }
        return entity;
    }

    public async Task<List<TipoEventoDto>> FindByName(string nome)
    {
        var list = await _context.TipoEvento
            .Where(x => x.Nome.ToLower().Contains(nome.ToLower()))
            .ToListAsync();

        return list.Select(x => ConvertToDto(x)).ToList();
    }

    private TipoEventoDto ConvertToDto(TipoEventoModel entity)
    {
        return new TipoEventoDto
        {
            Id = entity.Id,
            Nome = entity.Nome,
        };
    }

    private async Task<TipoEventoModel> ConvertToEntity(TipoEventoDto dto)
    {
        return new TipoEventoModel
        {
            Id = dto.Id ?? 0,
            Nome = dto.Nome
        };
    }
}